docs/how_to_guide/plot_05_transformer_basis.md

```diff
@@ -0,0 +1,183 @@
+
+Instantiating a [`TransformerBasis`](nemos.basis._transformer_basis.TransformerBasis) can be done either by using the constructor directly or with [`Basis.to_transformer()`](nemos.basis._basis.Basis.to_transformer):
+
+
+```{code-cell} ipython3
+bas = nmo.basis.RaisedCosineLinearConv(5, window_size=5)
+
+# initalize using the constructor
+trans_bas = nmo.basis.TransformerBasis(bas)
+
+# equivalent initialization via "to_transformer"
+trans_bas = bas.to_transformer()
+
+```
+
+[`TransformerBasis`](nemos.basis._transformer_basis.TransformerBasis) provides convenient access to the underlying [`Basis`](nemos.basis._basis.Basis) object's attributes:
+
+
+```{code-cell} ipython3
+print(bas.n_basis_funcs, trans_bas.n_basis_funcs)
+```
+
+We can also set attributes of the underlying [`Basis`](nemos.basis._basis.Basis). Note that -- because [`TransformerBasis`](nemos.basis._transformer_basis.TransformerBasis) is created with a copy of the [`Basis`](nemos.basis._basis.Basis) object passed to it -- this does not change the original [`Basis`](nemos.basis._basis.Basis), nor does changing the original [`Basis`](nemos.basis._basis.Basis) modify the [`TransformerBasis`](nemos.basis._transformer_basis.TransformerBasis) we created:
+
+
+```{code-cell} ipython3
+trans_bas.n_basis_funcs = 10
+bas.n_basis_funcs = 100
+
+print(bas.n_basis_funcs, trans_bas.n_basis_funcs)
+```
+
+As with any `sckit-learn` transformer, the `TransformerBasis` implements `fit`, a preparation step, `transform`, the actual feature computation, and `fit_transform` which chains `fit` and `transform`. These methods comply with the `scikit-learn` input structure convention, and therefore they all accept a single 2D array.
+
+## Setting up the TransformerBasis
+
+At this point we have an object equipped with the correct methods, so now, all we have to do is concatenate the inputs into a unique array and call `fit_transform`, right? 
+
+```{code-cell} ipython3
+
+# reinstantiate the basis transformer for illustration porpuses
+composite_basis = counts_basis + speed_basis
+trans_bas = (composite_basis).to_transformer()
+
+# concatenate the inputs
+inp = np.concatenate([counts, speed[:, np.newaxis]], axis=1)
+print(inp.shape)
+
+try:
+    trans_bas.fit_transform(inp)
+except RuntimeError as e:
+    print(repr(e))
+```
+
+...Unfortunately, not yet. The problem is that the basis has never interacted with the two separate inputs, and therefore doesn't know which columns of `inp` should be processed by `count_basis` and which by `speed_basis`.
+
+You can provide this information by calling the `set_input_shape` method of the basis. 
+
+This can be called before or after the transformer basis is defined. The method extracts and stores the array shapes excluding the sample axis (which won't be affected in the concatenation).
```

Comment:
```suggestion
This can be called before or after the transformer basis is defined. The method extracts and stores the number of columns for each input. There are multiple ways to call this method:

- It directly accepts the input: `composite_basis.set_input_shape(counts, speed)`
...
```

It's too many lines to make a good looking suggestion, but I think this should be a list like the above, with only a single line per example (rather than a full code block, though ideally still with the copy button and syntax highlighting).

I don't think we need to show the `out = composite_basis.to_transformer().fit_transform(inp)` line in each of them, just show it at the end.

Similarly, should have an admonition (or another small paragraph?) mentioning that we are looking at all dimensions except the first (since the samples are irrelevant here), and show a 4-dimensional example.